### PR TITLE
Replace Pusher Beams with Rpush PN delivery ✂️ ✂️ 

### PR DESCRIPTION
**Pull request by fdoxyz**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

This PR:
- Removes the `mobile_notifications` FeatureFlag
- Removes Pusher Beams dependency + related logic
- Adds a page to the docs about Rpush explaining the role `ConsumerApp` and `Device` models play in delivering Push Notifications to (mobile) apps.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/68

## QA Instructions, Screenshots, Recordings

1. Boot app locally and check out `/admin/apps/consumer_apps`
2. CRUD operations should work on `ConsumerApp` records
3. The official Forem iOS app should appear visible but **not operational**
4. If you add `RPUSH_IOS_PEM=anything` to your `.env` file and reboot your local site then **it should appear as operational** (because it detects the credentials from the ENV variable)

Screenshot of PN delivery working in Forem Cloud instances (not just [BHC](https://community.benhalpern.com))

Image from iOS

### UI accessibility concerns?

N/A

## Added tests?

- [ ] No, and this is why: We already have test coverage and this PR removes FeatureFlag specific & legacy (Pusher Beams) tests

## [Forem core team only] How will this change be communicated?

- [x] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?

None other than make sure Devices start to appear as registered and PNs are being delivered

## [optional] What gif best describes this PR or how it makes you feel?

rocket cowboy

### CODE REVIEW

**Review by joshpuetz — APPROVED** [review]
✂️✂️✂️

File: app/views/admin/consumer_apps/index.html.erb [hunk 1]
```diff
@@ -4,7 +4,10 @@
   <h1 class="crayons-title mb-6">Consumer Apps</h1>
 
   <p>
-    The following are apps that will be able to receive Push Notifications from <%= SiteConfig.community_name %>.
+    Consumer apps are standalone mobile apps that users can use to browse <%= SiteConfig.community_name %> as an alternative to web browsers.
+  </p>
+  <p class="mt-3">
+    If the app is listed as <strong>operational</strong> it means <%= SiteConfig.community_name %> will be able to deliver Push Notifications directly to them.
```
Comment (rhymes) [inline app/views/admin/consumer_apps/index.html.erb:10]: Tiny fix:

```suggestion
    If the app is listed as <strong>operational</strong>, it means <%= SiteConfig.community_name %> will be able to deliver push notifications directly to users' devices.
```

File: docs/backend/push-notifications.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+title: Push Notifications
+---
+
+# Push Notification (PN) Delivery
```
Comment (rhymes) [inline docs/backend/push-notifications.md:5]:
```suggestion
# Push Notifications Delivery
```

File: docs/backend/push-notifications.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Forem instances rely on [Rpush](https://github.com/rpush/rpush) to deliver Push
+Notifications (PN's). This decision was heavily influenced by the intention to
+provide as much flexibility as possible to Creators. In order to do this Forem
+instances can register & configure a `ConsumerApp`.
```
Comment (rhymes) [inline docs/backend/push-notifications.md:10]:
```suggestion
Forem instances rely on [Rpush](https://github.com/rpush/rpush) to deliver push
notifications. This decision was heavily influenced by the desire to
provide as much flexibility as possible to Creators. In order to do this, Forem
instances can register and configure a `ConsumerApp`.
```

File: docs/backend/push-notifications.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+These apps represent (mobile) apps that users use to browse (consume content)
+the Forem instance. Registered users of Forem instances can register a `Device`
+associated to a `ConsumerApp`. With these three pieces we are able to deliver a
+PN to a user's registered physical device(s).
```
Comment (rhymes) [inline docs/backend/push-notifications.md:15]:
```suggestion
These consumer apps represent mobile applications that users use to browse and consume content
on a Forem. Authenticated users of a specific Forem instance can register a `Device`
associated to a `ConsumerApp`. With these pieces we are able to deliver
push notifications to users devices.
```

File: docs/backend/push-notifications.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+and require their secret credential to be provided via ENV variable.
+
+## Rpush Implementation
+
+We use Rpush's `rpush-redis` implementation (read
+[this thread](https://github.com/forem/forem/pull/12419/files#r564660917) for
+the reasons why), and this means **we don't rely on Postgres** for all of Rpush
+functionality (no DB migrations). All Rpush models are persisted in Redis. More
+details about how this works
+[here](https://github.com/rpush/rpush/wiki/Using-Redis).
```
Comment (rhymes) [inline docs/backend/push-notifications.md:28]: I'm not sure we should tell users what we don't use, let's just tell them what we use :) 

```suggestion
We use Rpush's `rpush-redis` implementation (read
[this thread](https://github.com/forem/forem/pull/12419/files#r564660917) for
the reasons why), hence all `Rpush` models are persisted in Redis. More
details about how this works
[here](https://github.com/rpush/rpush/wiki/Using-Redis).
```

**Review by rhymes — COMMENTED** [review]
Great work @fdoxyz! I suggested some changes in the review

**Comment by fdoxyz** [comment]
@rhymes  those edits all sound like great enhancements to me, thank you!

**Review by rhymes — APPROVED** [review]
LGTM!

**Pull request merged**